### PR TITLE
Infix to prefix programs in c++ and java

### LINKED ISSUE #990: Infix to prefix conversion in java and c++  using two stacks
This will convert any infix to prefix expression using two stacks

### PR DESCRIPTION
**Pull request by robotjellyzone**
Fixes #990 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- added program for infix to prefix in java
- added program for infix to prefix in c++

#### Languages Used:
- c++
- java

#### Files Added:
- infix_to_prefix.cpp
- infixConversion.java

#### Folder added 
-infix_to_prefix

### CODE REVIEW

**Comment by jainaman224** [comment]
Please file and folder name.

File: infix_to_prefix/infixConversion.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    * @param infix string input
+    */
+    static String infixToPrefix(String infix){
+    // stack for operators. 
+	Stack<Character> operators = new Stack<Character>(); 
+
+    // stack for operands. 
+	Stack<String> operands = new Stack<String>(); 
+    for (int i = 0; i < infix.length(); i++){
+    // If current character is an
```
Comment (jainaman224) [inline infix_to_prefix/infixConversion.java]: Fix indentation.

File: infix_to_prefix/infixConversion.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    * This method converts infix 
+    * expression to prefix expression. 
+    * @param infix string input
+    */
+    static String infixToPrefix(String infix){
+    // stack for operators. 
+	Stack<Character> operators = new Stack<Character>(); 
+
+    // stack for operands. 
+	Stack<String> operands = new Stack<String>();
```
Comment (jainaman224) [inline infix_to_prefix/infixConversion.java]: Fix indentation.

File: infix_to_prefix/infixConversion.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    // stack for operators. 
+	Stack<Character> operators = new Stack<Character>(); 
+
+    // stack for operands. 
+	Stack<String> operands = new Stack<String>(); 
+    for (int i = 0; i < infix.length(); i++){
+    // If current character is an 
+	// opening bracket, then 
+	// push into the operators stack. 
+		if (infix.charAt(i) == '(') {
```
Comment (jainaman224) [inline infix_to_prefix/infixConversion.java]: Fix indentation.

File: infix_to_prefix/infix_to_prefix.cpp [hunk 4]
```diff
@@ -0,0 +1,126 @@
+// Including Library
+#include <bits/stdc++.h> 
+using namespace std; 
+
+    /** Function to check if given character is
```
Comment (jainaman224) [inline infix_to_prefix/infix_to_prefix.cpp]: Fix indentation.

File: infix_to_prefix/infixConversion.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    /**
+      * This method converts infix
+      * expression to prefix expression.
+      * @param infix string input
+      */
+    static String infixToPrefix(String infix) {
+        // stack for operators.
+       Stack<Character> operators = new Stack<Character>();
```
Comment (amaaniqbal) [inline infix_to_prefix/infixConversion.java]: Indentation is still not proper at some places

File: infix_to_prefix/infix_to_prefix.cpp [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    /** Final prefix expression is
+        present in operands stack. **/
+    return operands.top();
+}
+
+// Driver code
+int main()
+{
+    string s;
+    cin>>s;
```
Comment (amaaniqbal) [inline infix_to_prefix/infix_to_prefix.cpp]: Add space around `>>`

**Pull request merged**